src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +26,42 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_ADDRESS);
+
+        if (arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS)
+                || (argMultimap.getValue(PREFIX_NAME).isEmpty()
+                && argMultimap.getValue(PREFIX_ADDRESS).isEmpty())) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        String trimmedArgs = "";
+        boolean isFindByName = argMultimap.getValue(PREFIX_NAME).isPresent();
+        boolean isFindByAddress = argMultimap.getValue(PREFIX_ADDRESS).isPresent();
+
+        if (isFindByName) {
+            trimmedArgs = argMultimap.getValue(PREFIX_NAME).get().trim();
+        } else if (isFindByAddress) {
+            trimmedArgs = argMultimap.getValue(PREFIX_ADDRESS).get().trim();
+        }
+
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
         String[] nameKeywords = trimmedArgs.split("\\s+");
+        return isFindByName ? new FindNameCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)))
+                : new FindAddressCommand(new AddressContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+    }
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
```

Comment:
We should be able to use count() here since the stream is finite and we're not using it again.